core/src/toga/fonts.py

```diff
@@ -55,51 +66,45 @@ def __str__(self) -> str:
         return f"{self.family} {size}{weight}{variant}{style}"
 
     @staticmethod
-    def register(family, path, weight=NORMAL, style=NORMAL, variant=NORMAL):
-        """Registers a file-based font with its family name, style, variant
-        and weight. When invalid values for style, variant or weight are passed,
-        ``NORMAL`` will be used.
+    def register(family, path, *, weight=NORMAL, style=NORMAL, variant=NORMAL):
+        """Registers a file-based font.
+
+        **Note:** This is not currently supported on macOS or iOS.
+
+        When invalid values for style, variant or weight are passed, ``NORMAL`` will be
+        used.
 
-        When a font file includes multiple font weight/style/etc, each variant must be
-        registered separately::
+        When a font includes multiple weights, styles or variants, each one must be
+        registered separately, even if they're stored in the same file::
+
+            from toga.style.pack import BOLD
 
             # Register a simple regular font
-            Font.register("Font Awesome 5 Free Solid", "resources/Font Awesome 5 Free-Solid-900.otf")
+            Font.register(
+                "Font Awesome 5 Free Solid", "Font Awesome 5 Free-Solid-900.otf"
+            )
 
             # Register a regular and bold font, contained in separate font files
-            Font.register("Roboto", "resources/Roboto-Regular.ttf")
-            Font.register("Roboto", "resources/Roboto-Bold.ttf", weight=Font.BOLD)
-
-            # Register a single font file that contains both # a regular and bold weight
-            Font.register("Bahnschrift", "resources/Bahnschrift.ttf")
-            Font.register("Bahnschrift", "resources/Bahnschrift.ttf", weight=Font.BOLD)
-
-        :param family: The font family name. This is the name that can be referenced in
-            style definitions.
-        :param path: The path to the font file.
-        :param weight: The font weight. Default value is ``NORMAL``.
-        :param style: The font style. Default value is ``NORMAL``.
-        :param variant: The font variant. Default value is ``NORMAL``.
+            Font.register("Roboto", "Roboto-Regular.ttf")
+            Font.register("Roboto", "Roboto-Bold.ttf", weight=BOLD)
+
+            # Register a single font file that contains both a regular and bold weight
+            Font.register("Bahnschrift", "Bahnschrift.ttf")
+            Font.register("Bahnschrift", "Bahnschrift.ttf", weight=BOLD)
+
+        Parameters are the same as in the ``Font`` constructor above, with the addition
+        of:
```

Comment:
Does this work with autodoc? While there would be repetition in the definitions, I'm not sure I'm overly concerned with the repetition vs having all the details you need right in front of you at point of usage.

Reply:
(By "work" I mean does the final rendering make sense - it clearly "works" in the sense that it doesn't raise Sphinx errors)

Reply:
The rendering did make sense, but the repetition is fairly small, so I think you're right that it's outweighed by the advantage of making the method documentation self-contained. Fixed.